network/api_test.go

```diff
@@ -193,13 +210,18 @@ func TestCreateNetworkError(t *testing.T) {
 		t.Fatal(err)
 	}
 	defer response.Body.Close()
-	body, err := ioutil.ReadAll(response.Body)
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	if response.StatusCode != http.StatusInternalServerError {
 		t.Fatalf("Expected 500, got %d\n", response.StatusCode)
 	}
-	if string(body) != "{\"Err\":\"I CAN HAZ ERRORZ\"}\n" {
-		t.Fatalf("Expected %s, got %s\n", "{\"Err\":\"I CAN HAZ ERRORZ\"}\n", string(body))
+
+	expectedBody := "{\"Err\":\"I CAN HAZ ERRORZ\"}\n"
+	if string(body) != expectedBody {
```

Comment:
This test actually failed due to what has been fixed in https://github.com/Scalingo/go-handlers/pull/37

Reply:
Nitpick: I'm not sure to understand the reason of the failure. Would you like to explain a bit please ?

Reply:
The body returned by the call to the endpoint contained `{\"Err\":\"I CAN HAZ ERRORZ\"}\nI CAN HAZ ERRORZ`. Because go-handlers detected the type as non-JSON, and go-handlers didn't took into account that the body was already written.